### PR TITLE
Fix undefined behavior in s390x aes-gcm/ccm

### PR DESCRIPTION
**Pull request by p-steuer**
Fixes unaligned-access ub. s390 supports unaligned access, but ubsan will error.

### CODE REVIEW

File: crypto/evp/e_aes.c [hunk 1]
```diff
@@ -2291,7 +2307,7 @@ static int s390x_aes_ccm_ctrl(EVP_CIPHER_CTX *c, int type, int arg, void *ptr)
         memcpy(buf, ptr, arg);
         cctx->aes.ccm.tls_aad_len = arg;
 
-        len = *(uint16_t *)(buf + arg - 2);
+	len = buf[arg - 2] << 8 | buf[arg - 1];
```
Comment (dot-asm) [inline crypto/evp/e_aes.c]: Formatting nit. Expand tab.

Comment (p-steuer) [reply crypto/evp/e_aes.c]: oops sorry. fixed.

**Comment by richsalz** [comment]
reconfirm the whitespace. :)  @dot-asm let me know if you want me to merge.

**Comment by dot-asm** [comment]
Go ahead...

**Comment by richsalz** [comment]
merged, thanks for the fix!

**Pull request closed without merging**